huaweicloud/services/evs/resource_huaweicloud_evs_volume.go

```diff
@@ -208,22 +231,35 @@ func resourceEvsVolumeCreate(ctx context.Context, d *schema.ResourceData, meta i
 	if err != nil {
 		return fmtp.DiagErrorf("Error creating HuaweiCloud EVS volume: %s", err)
 	}
-	logp.Printf("[INFO] Volume Job ID: %s", job.JobID)
-
-	logp.Printf("[DEBUG] Waiting for the EVS volume to become available, the job ID is %s.", job.JobID)
-	stateConf := &resource.StateChangeConf{
-		Target:     []string{"SUCCESS"},
-		Refresh:    jobRefreshFunc(evsV1Client, job.JobID),
-		Timeout:    d.Timeout(schema.TimeoutCreate),
-		MinTimeout: 3 * time.Second,
+	logp.Printf("[INFO] The list of volume IDs is: %s", job.VolumeIDs)
+	if len(job.VolumeIDs) < 1 {
+		return fmtp.DiagErrorf("The volume ID was not included in the response to the request to create the volume.")
 	}
+	d.SetId(job.VolumeIDs[0])
 
-	resp, err := stateConf.WaitForStateContext(ctx)
-	if err != nil {
-		return fmtp.DiagErrorf("Error waiting for the creation of EVS volume (%s) to complete: %s", d.Id(), err)
+	// If charging mode is PrePaid, wait for the order to be completed.
+	if strings.EqualFold(d.Get("charging_mode").(string), "prePaid") {
+		err = common.WaitOrderComplete(ctx, d, config, job.OrderID)
+		if err != nil {
+			return fmtp.DiagErrorf("[DEBUG] Error the order is not completed while creating EVS volume (%s): %#v",
+				d.Id(), err)
+		}
+	} else {
```

Comment:
Is it confirmed that the disk is available when the order is complete?

Reply:
done